service/src/main/kotlin/io/provenance/explorer/service/async/ScheduledTaskService.kt

```diff
@@ -445,32 +439,25 @@ class ScheduledTaskService(
     }
 
     @Scheduled(initialDelay = 5000L, fixedDelay = 5000L)
-    fun startAccountProcess() = runBlocking {
-        ProcessQueueRecord.reset(ProcessQueueType.ACCOUNT)
-        val producer = startAccountProcess()
-        repeat(5) { accountProcessor(producer) }
+    fun startAccountProcess() {
+        processAccountRecords()
     }
 
-    @OptIn(ExperimentalCoroutinesApi::class)
-    fun CoroutineScope.startAccountProcess() = produce {
-        while (true) {
-            ProcessQueueRecord.findByType(ProcessQueueType.ACCOUNT).firstOrNull()?.let {
+    fun processAccountRecords() {
+        ProcessQueueRecord.reset(ProcessQueueType.ACCOUNT)
+        val records = ProcessQueueRecord.findByType(ProcessQueueType.ACCOUNT)
+        runBlocking {
+            for (record in records) {
                 try {
-                    transaction { it.apply { this.processing = true } }
-                    send(it.processValue)
+                    transaction { record.apply { this.processing = true } }
+                    accountService.updateTokenCounts(record.processValue)
```

Comment:
I see now - `updateTokenCounts` is suspended. I think you'll want to do the following because as it stands right now. the subsequent line is ran before `updateTokenCounts` completes, and also whether it throws an exception or not.

```suggestion
                    accountService.updateTokenCounts(record.processValue).await
```

Reply:
I updated my suggested change above so that there's no blocking calls inside of the `runBlocking`, even this isn't idiomatic, but at least it's correct from a concurrency perspective.

With this, the `runBlocking` on `449` can be removed.

Reply:
Yep, good catch.  I was hoping not to use runBlocking at all, but all the grpc client methods use suspend.  I decided I would look into why this pattern is used instead of change it.

Reply:
It's a great model, but requires up and down the stack to adhere to it. In this case, the jdbc driver is thread based calls so it requires constantly bridging blocking and nonblocking code.